opentelemetry-sdk/src/opentelemetry/sdk/logs/__init__.py

```diff
@@ -0,0 +1,181 @@
+        trace_id: Optional[int] = None,
+        span_id: Optional[int] = None,
+        trace_flags: Optional[TraceFlags] = None,
+        severity_text: Optional[str] = None,
+        severity_number: Optional[SeverityNumber] = None,
+        name: Optional[str] = None,
+        body: Optional[str] = None,
+        resource: Optional[Resource] = None,
+        attributes: Optional[Attributes] = None,
+    ):
+        self.timestamp = timestamp
+        self.trace_id = trace_id
+        self.span_id = span_id
+        self.trace_flags = trace_flags
+        self.severity_text = severity_text
+        self.severity_number = severity_number
+        self.name = name
+        self.body = body
+        self.resource = resource
+        self.attributes = attributes
+
+    def __eq__(self, other: object) -> bool:
+        if not isinstance(other, LogRecord):
+            return NotImplemented
+        return (
+            self.timestamp == other.timestamp
+            and self.trace_id == other.trace_id
+            and self.span_id == other.span_id
+            and self.trace_flags == other.trace_flags
+            and self.severity_text == other.severity_text
+            and self.severity_number == other.severity_number
+            and self.name == other.name
+            and self.body == other.body
+            and self.resource == other.resource
+            and self.attributes == other.attributes
+        )
+
+
+class LogData:
+    """Readable LogRecord data plus associated InstrumentationLibrary."""
+
+    def __init__(
+        self, log_record: LogRecord, instrumentation_info: InstrumentationInfo,
+    ):
+        self.log_record = log_record
+        self.instrumentation_info = instrumentation_info
+
+
+class LogProcessor(abc.ABC):
+    """Interface to hook the log record emitting action.
+
+    Log processors can be registered directly using
+    :func:`LogEmitterProvider.add_log_processor` and they are invoked
+    in the same order as they were registered.
+    """
+
+    @abc.abstractmethod
+    def emit(self, log_data: LogData):
+        """Emits the `LogData`"""
+
+    @abc.abstractmethod
+    def shutdown(self):
+        """Called when a :class:`opentelemetry.sdk.logs.LogEmitter` is shutdown"""
+
+    @abc.abstractmethod
+    def force_flush(self, timeout_millis: int = 30000):
+        """Export all the received logs to the configured Exporter that have not yet
+        been exported.
+
+        Args:
+            timeout_millis: The maximum amount of time to wait for logs to be
+                exported.
+
+        Returns:
+            False if the timeout is exceeded, True otherwise.
+        """
+
+
+class LogEmitter:
+    # TODO: Add multi_log_processor
+    def __init__(
+        self, resource: Resource, instrumentation_info: InstrumentationInfo,
+    ):
+        self._resource = resource
+        self._instrumentation_info = instrumentation_info
+
+    def emit(self, record: LogRecord):
```

Comment:
Should this be an abstract method?

Reply:
No, this will be called by the logging handler. It prepares LogData (from LogRecord & Instrumentation info) and will make a call to LogProcessors.